sched/queue/queue.go

```diff
@@ -0,0 +1,40 @@
+package queue
+
+import (
+	"github.com/scootdev/scoot/sched"
+	"time"
+)
+
+// Note: queued jobs can be read straight out of a channel that should
+// be returned along with the creator of a Queue
+type Queue interface {
```

Comment:
I'm also super interested in the Dequeue part Peek() and Dequeue() because there has to be a two way handshake there, to prevent data loss during crashes
- scheduler give me work, 
- once's its scheduled work and started the saga return i successfully started this
- then its safe to dequeue